roles/jd-client/src/lib/template_receiver/message_handler.rs

```diff
@@ -43,6 +43,17 @@ impl ParseServerTemplateDistributionMessages for TemplateRx {
         &mut self,
         _m: RequestTransactionDataError,
     ) -> Result<SendTo, Error> {
-        todo!()
+        let m = RequestTransactionDataError {
+            template_id: _m.template_id,
+            error_code: _m.error_code.into_static(),
+        };
+
+        match std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code") {
+            "template-id-not-found" => Err(Error::NoValidTemplate),
+            "stale-template-id" => Ok(SendTo::None(Some(
+                TemplateDistribution::RequestTransactionDataError(m),
+            ))),
+            _ => Err(Error::NoValidTemplate),
```

Comment:
since we match string here would be useful to log which is the string that is not parsable. So i suggest to use somthing like `Error::NoValidTemplate(error-message)`

Reply:
Done here 9693822b16924f744c31fc8d42fffed94f67776c